presto-main/src/main/java/com/facebook/presto/tdigest/Sort.java

```diff
@@ -0,0 +1,482 @@
+/*
+ * Licensed to Ted Dunning under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tdigest;
+
+import java.util.Random;
+
+/**
+ * Static sorting methods
+ */
+public class Sort {
+    private static final Random prng = new Random(); // for choosing pivots during quicksort
+    /**
+     * Quick sort using an index array.  On return,
+     * values[order[i]] is in order as i goes 0..values.length
+     *
+     * @param order  Indexes into values
+     * @param values The values to sort.
+     */
+    @SuppressWarnings("WeakerAccess")
```

Comment:
Curious: What are these annotations?

Reply:
Copied straight off from existing implementation. I removed the `WeakerAccess` ones which made no difference, only kept `SameParameterValue`.